File: zenox/auto_tasks/client_stats.py
import discord
from zenox.bot.bot import Zenox
from zenox.db.mongodb import DB
from datetime import datetime
from zenox.static.embeds import Embed
from zenox.static.utils import send_webhook

class client_stats:
    
    @classmethod
    def __get_embed(self, guild_count: int, user_count) -> Embed:
        desc = f"Guilds: {guild_count}\nUsers: {user_count}"
        embed = Embed(
            locale=discord.Locale.american_english,
            color=0xf189ff,
            title="Client Stats",
            description=desc
        )
        return embed

    @classmethod
    async def execute(self, client: Zenox):
        try:
            member_count = 0
            guild_count = len(client.guilds)

            for guild in client.guilds:
                member_count += guild.member_count
                DB.guilds.update_one({"id": guild.id}, {"$set": {"memberCount": guild.member_count}})

            DB.const.update_one(
                {"_id": "guild_growth"},
                {"$push": {"growth": {"date": datetime.now(), "guilds": guild_count}}},
                upsert=True
            )
            DB.const.update_one(
                {"_id": "guild_user_growth"},
                {"$push": {"growth": {"date": datetime.now(), "users": member_count}}},
                upsert=True
            )
            embed = self.__get_embed(guild_count, member_count)
            await send_webhook(client.log_webhook_url, username="Client Stats Task", embed=embed)
        except Exception as e:
            client.capture_exception(e)

File: src/auto_tasks/client_stats.ts
import { Zenox } from '../bot/bot';
import { DB } from '../db/mongodb';
import { Embed } from '../static/embeds';
import { sendWebhook } from '../static/utils';
import { log } from '../utils/logger';

export class ClientStats {
  private static getEmbed(guildCount: number, userCount: number): Embed {
    const desc = `Guilds: ${guildCount}\nUsers: ${userCount}`;
    const embed = new Embed(
      'en-US' as any,
      0xf189ff,
      "Client Stats",
      desc
    );
    return embed;
  }

  static async execute(client: Zenox): Promise<void> {
    log.separator('CLIENT STATS TASK');
    log.info('Starting client stats collection...');
    
    try {
      let memberCount = 0;
      const guildCount = client.guilds.cache.size;

      log.info(`Processing ${guildCount} guilds...`);
      
      // Update member counts for each guild
      for (const guild of client.guilds.cache.values()) {
        memberCount += guild.memberCount;
        DB.guilds.updateOne(
          { id: guild.id },
          { $set: { memberCount: guild.memberCount } }
        );
      }

      // Update growth statistics
      DB.const.updateOne(
        { _id: "guild_growth" },
        { 
          $push: { 
            growth: { 
              date: new Date(), 
              guilds: guildCount 
            } 
          } 
        },
        { upsert: true }
      );

      DB.const.updateOne(
        { _id: "guild_user_growth" },
        { 
          $push: { 
            growth: { 
              date: new Date(), 
              users: memberCount 
            } 
          } 
        },
        { upsert: true }
      );

      // Log statistics
      log.table('Client Statistics', {
        'Total Guilds': guildCount,
        'Total Users': memberCount,
        'Average Users per Guild': Math.round(memberCount / guildCount)
      });

      // Send webhook with stats
      const embed = ClientStats.getEmbed(guildCount, memberCount);
      await sendWebhook(
        client.logWebhookUrl!,
        "Client Stats Task",
        undefined,
        embed
      );

      log.success('Client stats collected and sent successfully');
      log.separator('CLIENT STATS COMPLETE');
    } catch (error) {
      log.error('Error collecting client stats', error);
      client.captureException(error as Error);
    }
  }
}